opensearch/src/http/aws_auth.rs

```diff
@@ -0,0 +1,82 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+use std::time::SystemTime;
+
+use aws_sigv4::{
+    http_request::{
+        sign, PayloadChecksumKind, PercentEncodingMode, SignableBody, SignableRequest,
+        SignatureLocation, SigningParams, SigningSettings,
+    },
+    signing_params::BuildError,
+};
+use aws_types::{
+    credentials::{ProvideCredentials, SharedCredentialsProvider},
+    region::Region,
+    Credentials,
+};
+use reqwest::Request;
+
+const SERVICE_NAME: &str = "es";
+
+fn get_signing_params<'a>(
+    credentials: &'a Credentials,
+    region: &'a Region,
+) -> Result<SigningParams<'a>, BuildError> {
+    let mut settings = SigningSettings::default();
+    settings.percent_encoding_mode = PercentEncodingMode::Double;
+    settings.payload_checksum_kind = PayloadChecksumKind::NoHeader;
+    settings.signature_location = SignatureLocation::Headers;
+    settings.expires_in = None;
+
+    let mut builder = SigningParams::builder()
+        .access_key(credentials.access_key_id())
+        .secret_key(credentials.secret_access_key())
+        .service_name(SERVICE_NAME)
+        .region(region.as_ref())
+        .time(SystemTime::now())
+        .settings(settings);
+
+    if let Some(session_token) = credentials.session_token() {
+        builder = builder.security_token(session_token);
```

Comment:
this could be replaced with `builder.set_security_token(credentials.session_token())` ?

Reply:
alternatively should this output an error message if this is `None`?

Reply:
Yes, will replace it 👍 The `None` state of session_token is fine, as permanent credentials won't have one, while temporary ones such as EC2 instance credentials etc will have one.